packages/flutter_tools/lib/src/build_system/targets/dart.dart

```diff
@@ -140,6 +124,28 @@ class CopyFlutterBundle extends Target {
   ];
 }
 
+/// A helper function to copy an [assetBundle] into an [environment]'s output directory,
+/// plus an optional [pathSuffix]
+Future<void> copyAssets(AssetBundle assetBundle, Environment environment, [String pathSuffix = '']) async {
+  final Pool pool = Pool(64);
+  await Future.wait<void>(
+    assetBundle.entries.entries.map<Future<void>>((MapEntry<String, DevFSContent> entry) async {
+      final PoolResource resource = await pool.request();
+      try {
+        final File file = fs.file(fs.path.join(environment.outputDir.path, pathSuffix, entry.key));
+        file.parent.createSync(recursive: true);
```

Comment:
For the pool to be effective, wouldn't you want to stick with async I/O only?

Reply:
No, sync methods create and dispose the fd immediately